### DISCUSSION TITLE
Home Assistant Integration: Keine Integration von "Homematic(IP) Local" möglich.

### DISCUSSION THREAD

**Discussion by JackNicholsen** [OpenCCU/OpenCCU#1901] [🇩🇪 Fragen&Antworten]
Hallo,
Homematic(IP) Local lief bei mir schon mal.
<img width="212" alt="image" src="https://user-images.githubusercontent.com/28046447/182643855-1013c6e6-f7bc-490a-9040-0a33b2737b02.png">
Ab irgend einem Zeitpunkt aber nicht mehr. Nachdem verschiedene Maßnahmen nicht funktioniert haben, habe ich die Integration gelöscht und versucht, sie neu einzurichten. Das klappt jetzt leider nicht mehr.
Hier die Beschreibung des aktuellen Stands: In HACS ist nur die aktuelle Version zu sehen:
<img width="459" alt="image" src="https://user-images.githubusercontent.com/28046447/182632231-376674f0-de95-4c54-8512-67f7c1168a1f.png">
Aber bei den Integrationen bekomme ich zwei unterschiedliche Versionen angezeigt:
<img width="381" alt="image" src="https://user-images.githubusercontent.com/28046447/182632447-887f5641-65c7-4084-84c1-7ce7b43e6fa3.png">
Bei der oberen Version (mit Icon) erhalte ich die Fehlermeldung:
<img width="449" alt="image" src="https://user-images.githubusercontent.com/28046447/182632683-5ea88f1f-d218-4ce0-9703-de40e64cfe8c.png">
Bei der unteren Version (mit dem "icon not available") erhalte ich zwar die korrekte Eingabemaske zum Ausfüllen der Parameter, aber eine Verbindung zu RaspberryMatic ist nicht möglich:
<img width="312" alt="image" src="https://user-images.githubusercontent.com/28046447/182633601-2ea80e14-2e83-4c4f-8298-a20597f465c6.png">

Ich bin von Anfang an dabei (vom der ersten RaspberryMatic-AddOn und später der ersten Homematic(IP) Local-Integration) und habe immer alle Updates eingespielt (fast immer täglich). Ich habe in Erinnerung, dass Homematic-(IP) zwischendurch einmal umgestellt wurde (Von "Custom_Homematic" auf das jetzige "Homematic local") und vermute, dass mein Problem damit zusammen hängt. Ich habe deshalb ein früheres Update eingespielt, aber auch dort funktioniert es nicht mehr. Vermutlich existiert irgendwo eine Einstellung oder eine Datei, welche weder durch das Löschen der Integration, als auch durch das Restore eines Backups nicht gelöscht wird. So existierte immer noch eine Datei "config/customize.yaml" mit der Konfiguration aus RaspberryMatic. Ein Löschen und Neustart brachte keinen Erfolg.

Eine Neuinstallation von Home Assistant würde sicherlich den Erfolg bringen. Weil ich eine recht umfangreiche Installation mit vielen Schnittstellen und Automatisierungen besitze, dauert das aber mehrere Tage. Ich bin deshab daran interessiert, das Problem ohne Neuinstallation zu lösen, und bitte um Hilfe!!!

Vielen Dank im Voraus!

**Comment by Baxxy13** [comment]
Die 0.0.0.0 als Callback Adresse sieht mir irgendwie falsch aus. Ich lasse das immer "leer".
Ansonsten mal das custom repository aus HACS entfernen (vorher die Integration entfernen) HA komplett neu starten und neu beginnen. Vielleicht hilft's.

**Reply by JackNicholsen** [reply]
Die "0.0.0.0" als Callback-Adresse trage ich nicht ein. Ich lasse den Eintrag leer. Der Eintrag erscheint, wenn die Fehlermeldung zurück kommt. Ich habe alle Eintragungen wie vorgeschrieben eingetragen und: Es hat ja schon mal funktioniert (siehe Screenshot in der Beschreibung).
Erst einmal vielen Dank, für die promte Reaktion und dass Du Dir Gedanken gemacht hast!
Das Custom Repository hatte ich entfernt. Es hat aber nichts genützt. Ich vermute deshalb (wie beschrieben), dass irgendein Überbleibsel vorhanden ist, welches beim Löschen des Repositories und auch beim Wiedereinspielen eines alten Backups nicht gelöscht wird und stört. Ein Hinweis darauf ist ja (wie beschrieben) die Existenz der Datei "config/customize.yaml", die ja eigentlich nach dem Löschen der Integration und des Repositories nicht mehr vorhanden sein dürfte.

**Reply by JackNicholsen** [reply]
Ich habe jetzt ein altes Backup vom 6.1.2022 eingespielt, wo Homematic(IP) local noch nie installiert war, habe HASS.IO und RaspberryMatic aktualisiert (weil es dafür notwendig ist), für HM(IP) das Repository eingetragen und installiert. Trotzdem bleibe ich bei der Integration hängen. Schlimm ist, dass jedesmal, wenn ich so etwas ausprobiere, meine Automatisierungen (Licht einschalten, Rollos herunter- herauffahren) usw. nicht funktionieren und sich meine Failienmitglieder (berechtigt) beschweren.

**Reply by JackNicholsen** [reply]
Mit HM(IP) arbeite ich seit dem 11.1.2022. Der Screenshot (siehe oben), in welchem noch alles funktioniert, stammt von Anfang Mai 2022. Also muss die Änderung danach passiert sein, welche die Probleme verursacht.

**Reply by JackNicholsen** [reply]
Die Ursache für meine fehlerhafte Konfiguration ist anscheinend tatsächlich in der Umbenennung von hahm in "Homatic(ip) local" am 29.03.2022 anlässlich des Übergangs vom Release 0.38.5 zum Release 1.0 zu suchen (siehe Datei "changelog.txt" im Quellcode):
<img width="325" alt="image" src="https://user-images.githubusercontent.com/28046447/182976982-7bfb2a77-ab9b-47db-a111-a63a10361afc.png">
Das hat die HACS-Konfiguration aus dem Tritt gebracht. Ich hätte damals die alte Version zunächst löschen sollen, ehe ich die neue installiere.
Mit dem Löschen des Downloads und des Repositiories von "Homatic(ip) local" verschwinden auch alle Einträge im HASS.IO. Trotzdem wird mir noch ein "Homatic(ip) local" ohne Icon zur Integration angeboten, welche aber nicht funktioniert. Die sich öffnende Eingabemaske stammt aus dem File "strings.json", welches aus dem hahm-Repository stammt. Dieses Repository ist in der HACS-Konfigurationsdatei "/config/custom_components/.storage/hacs_repositories" sowie im Verzeichnis "/config/custom_components/.storage/hacs/" aber nicht mehr zu finden. Also funktioniert das gesteuerte Löschen auch nicht mehr. Ich habe jetzt die Verzeichnise "/config/hahm/" (Funktion: cache) zu "/config/hahm_old" und "/config/custom_components/hahm" zu "/config/custom_components/hahm_old" umbenannt, damit sie nicht mehr gefunden werden, und HASS.IO neu gestartet.
<img width="195" alt="image" src="https://user-images.githubusercontent.com/28046447/182978571-981d4b06-d69c-4daf-8ca5-f074d8670f08.png">

<img width="221" alt="image" src="https://user-images.githubusercontent.com/28046447/182978497-8f9c9c86-b127-4c62-a8ef-203b5caa7ace.png">
Trotzdem wird mir die nicht funktionierende Integration (also die ohne Icon) noch angeboten:
<img width="380" alt="image" src="https://user-images.githubusercontent.com/28046447/182978131-86700a77-1aa2-45ab-b853-fcfa7e2c52cf.png">

<img width="304" alt="image" src="https://user-images.githubusercontent.com/28046447/182978239-0e33aa38-6e7d-4f45-a01e-d592fcb78fed.png">

<img width="313" alt="image" src="https://user-images.githubusercontent.com/28046447/182978286-a868ef86-0fd1-42ec-9fa7-a95a6b3a85e8.png">

<img width="304" alt="image" src="https://user-images.githubusercontent.com/28046447/182978318-0f45376c-c2f8-4829-a472-42c36c8586db.png">
Lösche ich die beiden Verzeichnisse ganz, dann wird mir die Integration (ohne Icon - wie im Screenshot oben) trotzdem angezeigt. Allerdings ist jetzt kein Code mehr hinterlegt (ist ja gelöscht) und die bekannte Fehlermeldung kommt:
<img width="461" alt="image" src="https://user-images.githubusercontent.com/28046447/182980662-73e03f36-bfa2-4b24-a89f-137e9c47a3f0.png">
Wer weiß, woher das Angebot zur Integration kommt und kann mir helfen? Ich kenne mich in Home Assistant leider nicht so gut aus. Es muss doch Leute geben, welche beim Beta-Test dabei waren und ebenfalls vor dem gleichen Problem stehen, wie ich! Bitte helft mir!

**Reply by JackNicholsen** [reply]
Nach meinen Maßnahmen im vorherigen Blog lässt sich das "Homematic(IP) Lokal"-Repository (dieses Mal das mit dem Icon!) installieren. Die Integration ohne Icon ist nach dem Herunterladen des Repositories verschwunden.
<img width="391" alt="image" src="https://user-images.githubusercontent.com/28046447/182982214-cfce0749-945e-426c-9de6-1fc066f11f25.png">
Die "hahm"-Verzeichnisse sind nicht wieder aufgetaucht, dafür ein Verzeichnis "/config/custom_components/homematicip_local"
<img width="198" alt="image" src="https://user-images.githubusercontent.com/28046447/182982807-11d89a23-9448-4c16-a05b-feb6e0d47db5.png">
Das ist schon mal ein Fortschritt gegenüber dem Stand davor. Meine Maßnahmen waren also zumindest zum Teil erfolgreich.
Allerdings ist immer noch keine Verbindung zu Raspberrymatic möglich:
<img width="305" alt="image" src="https://user-images.githubusercontent.com/28046447/182982552-96ada645-d136-4da7-a1d5-6fb8316de5bb.png">
Es ist ja auch noch nicht alles in Ordnung. Irgendwo schwirrt das "Homematic(IP) Lokal" (ohne Icon) immer noch herum. Ich benötige also trotzdem noch Eure Hilfe!

**Comment by EAFSL** [comment]
Hallo @JackNicholsen 
konntest du das Problem lösen?
Ich stehe aktuell genau vor demselben. 

Beste Grüße

**Reply by JackNicholsen** [reply]
Hallo EAFSL,

das Problem konnte ich nicht beseitigen, aber die Wirkung habe ich beseitigt. Es läuft wieder. Das Problem ist meine Schwiegermutter, die den RF-USB-Stick aus meinem Raspi entfernt hatte. Die Schwiegermutter ist noch da.

Da mein Raspi an einer sonst nicht zugänglichen Stelle stand, habe ich das Fehlen des USB nicht sofort gemerkt. Wenn es erst einmal läuft, ist leider nicht sofort ersichtlich, wenn jemand "den Stecker zieht". So habe ich das System neu installiert, was ziemlich viel Aufwand darstellt. Bei der Neuinstallation ist mir dann die Fehlermeldung wegen des fehlenden Sticks aufgefallen.

Inzwischen bin ich auch vom Raspi weg auf ein QNAP-NAS umgezogen und das steht neben mir. Da hat die Schwiegermutter keine Chance mehr!

Viele Grüße, Jacky

**Comment by DanielHoelder** [comment]
Guten Tag,
ich habe auch das Problem, dass hie Homematic(IP) local Integration das Raspberrymatic Addon nicht findet, bzw. nich nicht verbinden kann.

Nach meinem Verständnis hat RaspberryMatic den HmIP-RFUSB gefunden. Auszug aus dem Logfile:

```
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:00:14.0-2, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:00:14.0-2, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
```

Kann es sein, dass es daran liegt, dass ich an den Stick noch keine Geräte angelernt habe?
